File: ligature-http-in-memory/jvm/src/main/scala/dev/ligature/http/memory/LigatureHttpInMemory.scala
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

package dev.ligature.http.memory

import cats.effect.*
import com.comcast.ip4s.Port
import dev.ligature.inmemory.InMemoryLigature
import dev.ligature.http.{AuthMode, runLigature}
import dev.ligature.inmemory.createInMemoryLigature

case class LigatureConfig(
   authMode: AuthMode = AuthMode.None,
   port: Port = Port.fromInt(4200).get,
)

object MainLigatureHttp extends IOApp {
  def run(args: List[String]): IO[ExitCode] =
    createInMemoryLigature().use { instance =>
      val config = LigatureConfig()
      runLigature(instance, config.authMode, config.port)  
    }
}

File: ligature-http-in-memory/src/main/kotlin/dev/ligature/http/memory/LigatureHttpInMemory.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

package dev.ligature.http.memory

import io.ktor.server.engine.*
import io.ktor.server.netty.*

import dev.ligature.inmemory.InMemoryLigature
import dev.ligature.http.*

fun main() {
  val ligature = InMemoryLigature()
  embeddedServer(Netty, port = 4200, host = "0.0.0.0") {
    routes(ligature)
  }.start(wait = true)
}
